ldmud: 3.6.7 -> 3.6.8

cpu opened the pull request:
This branch:

* Applies some modernization that was [mentioned in this previous PR](https://github.com/NixOS/nixpkgs/pull/402169)
* Updates from 3.6.7 to 3.6.8

Here's the [ldmud 3.6.8 announcement post](https://groups.google.com/g/ldmud-talk/c/yCvuj6WWEBk/m/40BonmTqCwAJ), and the [change log](https://github.com/ldmud/ldmud/blob/b152e3aebb1fad6eff8a74dda1741d4e9add4447/HISTORY#L7-L61).

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

cpu committed:
ldmud: apply some small modernizations

Catches up with some best practices.

cpu committed:
ldmud: 3.6.7 -> 3.6.8

Notably this update allows dropping the libxm2 compat patch since the
fix has landed upstream.

See https://groups.google.com/g/ldmud-talk/c/yCvuj6WWEBk/m/40BonmTqCwAJ
for more information.

Defelo commented on pkgs/by-name/ld/ldmud/package.nix:
```diff
@@ -26,29 +26,29 @@
-    repo = "ldmud";
-    rev = version;
-    sha256 = "sha256-PkrjP7tSZMaj61Hsn++7+CumhqFPLbf0+eAI6afP9HA=";
+    repo = finalAttrs.pname;
```
```suggestion
    repo = "ldmud";
```

see #277994

Defelo commented on pkgs/by-name/ld/ldmud/package.nix:
```diff
@@ -26,29 +26,29 @@
-    rev = version;
-    sha256 = "sha256-PkrjP7tSZMaj61Hsn++7+CumhqFPLbf0+eAI6afP9HA=";
+    repo = finalAttrs.pname;
+    rev = finalAttrs.version;
```
```suggestion
    tag = finalAttrs.version;
```

Defelo replied:
Looks like the change hasn't been applied, it's still `rev` instead of `tag`.

Defelo commented on pkgs/by-name/ld/ldmud/package.nix:
```diff
@@ -26,29 +26,29 @@
-    sha256 = "sha256-PkrjP7tSZMaj61Hsn++7+CumhqFPLbf0+eAI6afP9HA=";
+    repo = finalAttrs.pname;
+    rev = finalAttrs.version;
+    sha256 = "sha256-ojOLM1vkuwuF0vXx6lCH0+OlyLkkOOnTJEUiZPpUhzo=";
```
```suggestion
    hash = "sha256-ojOLM1vkuwuF0vXx6lCH0+OlyLkkOOnTJEUiZPpUhzo=";
```

cpu replied:
Fixed.

cpu committed:
ldmud: fix MacOS iconv link time errors

The LDMud autotools build needs some help finding the right libiconv on
MacOS systems. Without nudging it the right direction with LDFLAGS the
system libiconv is used and produces missing symbol errors at link-time.

cpu commented:
> x86_64-darwin (sandbox = true)
> ❌ 2 packages failed to build:
> aarch64-darwin (sandbox = true)
> ❌ 2 packages failed to build:

I believe these will be fixed by https://github.com/NixOS/nixpkgs/pull/424458/commits/468fd1ddba67abf549c9ac9cc8d5facb970dd9cb

cpu commented:
Here's an updated report (_very cool tooling! Thank you @defelo :clap:_)

## `nixpkgs-review` result for [#424458](https://github.com/NixOS/nixpkgs/pull/424458)

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 424458`

Logs: https://github.com/cpu/nixpkgs-review-gha/actions/runs/16252116062


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>ldmud</li>
    <li>ldmud-full</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>ldmud</li>
    <li>ldmud-full</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>ldmud</li>
    <li>ldmud-full</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>ldmud</li>
    <li>ldmud-full</li>
  </ul>
</details>
